providers/supermicro/supermicro.go

```diff
@@ -303,8 +334,8 @@ func (c *Client) bmcQueryor(ctx context.Context) (bmcQueryor, error) {
 	}
 
 	model := strings.ToLower(queryor.deviceModel())
-	if !strings.HasPrefix(model, "x12") && !strings.HasPrefix(model, "x11") {
-		return nil, errors.Wrap(ErrModelUnsupported, "expected one of X11* or X12*, got:"+model)
+	if !strings.HasPrefix(model, "x13") && !strings.HasPrefix(model, "x12") && !strings.HasPrefix(model, "x11") {
```

Comment:
With your new `expected` function I don't think you'll actually get here if it's not `x11`, `x12`, or `x13`. I think the queryor nil check @ 301 covers you.